main.tf

```diff
@@ -24,6 +25,11 @@ resource "ibm_is_vpc" "vpc" {
   no_sg_acl_rules             = var.clean_default_sg_acl
 }
 
+data "ibm_is_vpc" "vpc" {
```

Comment:
A local will be required to determine which ibm_is_vpc.vpc is required

```hcl
locals {
  vpc_id = var.create_vpc ? var.existing_vpc_id : ibm_is_vpc.vpc.id
}
```

IF it is only the ID that is used, there may be no need to data resource lookup.

Additional validation is required to detect errors early (plan), so make sure that if create_vpc is false, then existing_vp_id MUST be passed.